src/points/ActivityCardSection.tsx

```diff
@@ -1,132 +1,122 @@
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import ActivityCard, { Props as ActivityCardProps } from 'src/points/ActivityCard'
+import { sortByAmountAndTitle } from 'src/points/cardSort'
+import { BottomSheetParams, PointsActivity } from 'src/points/types'
+import Colors from 'src/styles/colors'
 import { typeScale } from 'src/styles/fonts'
 import { Spacing } from 'src/styles/styles'
 
 interface Props {
+  pointsActivities: PointsActivity[]
   onCardPress: (bottomSheetDetails: BottomSheetParams) => void
-  pointsSections: PointsMetadata[]
 }
 
-export default function ActivityCardSection({ onCardPress, pointsSections }: Props) {
+export default function ActivityCardSection({ pointsActivities, onCardPress }: Props) {
   const { t } = useTranslation()
 
-  function makeSection(pointsMetadata: PointsMetadata, sectionIndex: number): React.ReactNode {
-    const pointsAmount = pointsMetadata.pointsAmount
-
-    const cards = pointsMetadata.activities
-      .filter((activity) => isPointsActivityId(activity.activityId))
-      .map((activity) => (
-        <ActivityCard
-          key={activity.activityId}
-          activityId={activity.activityId}
-          pointsAmount={pointsAmount}
-          onPress={onCardPress}
-        />
-      ))
-
-    // add the "more coming" card to the last section
-    if (sectionIndex === pointsSections.length - 1) {
-      cards.push(
-        <ActivityCard
-          key="more-coming"
-          activityId="more-coming"
-          pointsAmount={pointsAmount}
-          onPress={onCardPress}
-        />
-      )
+  function mapActivityToCardProps(activity: PointsActivity): ActivityCardProps {
+    switch (activity.activityId) {
+      case 'create-wallet':
+        return {
+          ...activity,
+          title: t('points.activityCards.createWallet.title'),
+          icon: <Celebration />,
+        }
+      case 'swap':
+        return {
+          ...activity,
+          title: t('points.activityCards.swap.title'),
+          icon: <SwapArrows />,
+          onPress: () =>
+            onCardPress({
+              ...activity,
+              title: t('points.activityCards.swap.bottomSheet.title'),
+              body: t('points.activityCards.swap.bottomSheet.body', {
+                pointsValue: activity.pointsAmount,
+              }),
+              cta: {
+                text: t('points.activityCards.swap.bottomSheet.cta'),
+                onPress: () => {
+                  navigate(Screens.SwapScreenWithBack)
+                },
+              },
+            }),
+        }
+      default:
+        // To catch any missing cases at compile time
+        const assertNever: never = activity.activityId
+        return assertNever
     }
+  }
 
-    if (!cards.length) {
-      return <View key={pointsAmount}></View>
-    }
+  const preparedActivities = pointsActivities.map(mapActivityToCardProps).sort(sortByAmountAndTitle)
```

Comment:
```suggestion
  const sortedActivities = pointsActivities.map(mapActivityToCardProps).sort(sortByAmountAndTitle)
```
nit: "prepared" makes me think too much of all our prepared transactions stuff....

Reply:
one thing that was nice about having this sorting in the selector is that we get memoization for free. of course the config is so small here that it doesn't really matter. but i wonder if we should do a useMemo here anyway...

Reply:
yes, sure!